pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -581,23 +581,19 @@ func (pl *dynamicResources) PreScore(ctx context.Context, cs *framework.CycleSta
 }
 
 func haveAllNodes(nodeNames []string, nodes []*v1.Node) bool {
+	if len(nodeNames) < len(nodes) {
+		return false
+	}
+	sort.Strings(nodeNames)
```

Comment:
`sort.Strings` sorts `nodeNames` in-place, which means it can change `podScheduling.Spec.PotentialNodes`. 
@pohly is `podScheduling.Spec.PotentialNodes` sorted? Should we avoid re-ordering it?

Reply:
We must be careful about updating `podScheduling.Spec.PotentialNodes`: if the object is the one from the informer cache, then it's shared and conceptually read-only.

If we find that sorting is crucial for good performance, then we can make it a requirement for these fields and enforce it through validation. That should also make validation more efficient: instead of having to build a set to find duplicates (as it does now), it can simply iterate and compare each entry with its successor.

That's an API break, but we have enough changes piled up already that we probably should do v1alpha2 with that change.

Reply:
I think we can add a TODO here first, wait until ```podScheduling.Spec.PotentialNodes``` are sorted. How about it？

Reply:
In my understanding we can't merge this PR until podScheduling.Spec.PotentialNodes are sorted. Current implementation sorts `podScheduling.Spec.PotentialNodes` which is quite dangerous according to @pohly

Reply:
OK. I will close this PR soon.thanks.

Reply:
@sourcelliu Can you create an issue about it, so we can return to it when podScheduling.Spec.PotentialNodes. Otherwise this work has a big chance to be forgotten.

Reply:
have you tried to make a copy of `podScheduling.Spec.PotentialNodes` and test it using `sets.Set`?